binder/src/androidTest/java/io/grpc/binder/internal/LeakSafeOneWayBinderTest.java

```diff
@@ -29,28 +30,34 @@
 public final class LeakSafeOneWayBinderTest {
 
   private LeakSafeOneWayBinder binder;
+  private FakeHandler handler = new FakeHandler();
 
-  private int transactionsHandled;
-  private int lastCode;
-  private Parcel lastParcel;
+  static class FakeHandler implements TransactionHandler {
+    int transactionsHandled;
+    int lastCode;
+    Parcel lastParcel;
```

Comment:
It changes the structure of the test, but should this be a list of value classes?

Then, you don't need to track the transaction count and it's more obvious when you're accessing the only vs the last transaction (you can even do `assertThat(handler.transactions).containsExactly(theSingleExpectedTransaction)`).

Reply:
That would be a nice improvement but isn't necessary for this PR. It would also be somewhat tricky to do because Parcels are mutable and technically not valid to touch after you onTransact() returns.